src/views/Browse/components/Filters.tsx

```diff
@@ -0,0 +1,63 @@
+import { faFile, faFolder, faSearch } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React, { useCallback, useMemo } from "react";
+
+import { BrowserFile, BrowserItem } from "../../../common/types";
+import { getHumanSize } from "../../../common/utils";
+import { ThreeWaySwitch } from "../../../components";
+import MultiRangeSlider from "../../../components/RangeSlider";
+import { TypeFilterValue, typeFilterValues, useFilter } from "../providers";
+
+export const Filters: React.FC<{ items: BrowserItem[] }> = ({ items }) => {
+  const { nameFilter, setNameFilter, sizeFilter, setSizeFilter, typeFilter, setTypeFilter } =
+    useFilter();
+
+  const formatSize = useCallback((v: number) => getHumanSize(v), []);
+
+  const sizeMinMax = useMemo(() => {
+    const sizes = (items.filter((i) => i.type === "file") as BrowserFile[]).map((i) => i.sizeBytes);
```

Comment:
was type inference broken for some reason? Why this type assertion?

Reply:
Yeah, typescript doesn't seem to be able to infer that the resulting array after `filter` is not a generic `BrowserItem` but must be `BrowserFile`.